esp/platform/application_config.cpp

```diff
@@ -89,6 +89,23 @@ static void copyDirectories(IPropertyTree *target, IPropertyTree *src)
     }
 }
 
+void addAuthDomains(const char *authtype, IPropertyTree *legacyEsp)
+{
+    if (isEmptyString(authtype)) //backward compatible
```

Comment:
If only AuthDomain/@authType changes, you can simplify this:

```
  if (isEmptyString(authtype))
      authtype = "AuthPerRequestOnly";

 VStringBuffer AuthDomains(
            "<AuthDomains>"
                "<AuthDomain authType='%s' clientSessionTimeoutMinutes='120' domainName='default'"
                    " invalidURLsAfterAuth='/esp/login' loginLogoURL='/esp/files/eclwatch/img/Loginlogo.png'"
                    " logonURL='/esp/files/Login.html' logoutURL='' serverSessionTimeoutMinutes='240'"
                    " unrestrictedResources='/favicon.ico,/esp/files/*,/esp/xslt/*'/>"
            "</AuthDomains>", authtype);
        legacyEsp->addPropTree("AuthDomains", createPTreeFromXMLString(AuthDomains));
```

Reply:
Fixed